### ISSUE TITLE
vector edges for ROIs

### ISSUE THREAD

**Issue by bendichter** [NeurodataWithoutBorders/pynwb#1122]
During visualization and analysis it is sometimes preferable to have an edge-wise vector representation of each ROI. This would generalize to a triangular mesh for 3D ROIs. It would be great if we could compute these from the masks. ROIs are non-convex, so we can't simply use a convex hull method. I'm not sure if this falls within scope for pynwb.

**Comment by oruebel** [comment]
Do you mean a contour plot to show the boundaries of ROIs? I.e., something like [marching squares](https://scikit-image.org/docs/dev/auto_examples/edges/plot_marching_cubes.html) (for 2D) or [marching cubes](https://scikit-image.org/docs/dev/auto_examples/edges/plot_contours.html) (for 3D) (see the links for corresponding examples).

**Comment by bendichter** [comment]
Yes this is perfect! I got the solution I was looking for with:

```python
from skimage import measure
import matplotlib.pyplot as plt

roi_index = 6
threshold=.01
img_mask = nwb.processing['Image-Segmentation'].data_interfaces['img_seg']['pln_seg']['image_mask'][roi_index]

x,y = zip(*measure.find_contours(img_mask, theshold)[0])
plt.plot(x, y)
```
Do you think we should include something like this in pynwb?

**Comment by oruebel** [comment]
That sounds reasonable. I think either as a separate function on the ophys module or on the corresponding container would make sense. I think I'd probably just do a function to get the geometry of the contours and leave the plotting up to the user. If you think we should include the plotting as well, then I think it would at least be useful to separate the computation of the geometry from the plotting, since folks will probably want to be able to get the geometry as well.

**Comment by bendichter** [comment]
I agree that the plotting should be left to the user. It's just a straightforward `plt.plot` call to get something reasonable, and we couldn't predict what they would want beyond that.
